ODIN_II/SRC/odin_util.cpp

```diff
@@ -909,6 +909,104 @@ void trim_string(char* string, const char* chars) {
     }
 }
 
+/*
+ * Tokenizes a string and returns an array where each element is a 
+ * token. The last element is set to 0
+*/
+char ** string_to_token_array(char * string)
+{
+    char ** arr = NULL;
+     
+    char * temp = strtok(string," ");
+    int i = 0;
+    int size;
+
+    while(temp != NULL)
+    {
+        size = strlen(temp);
+        arr = (char**)realloc(arr,sizeof (char*)*(size));
+            if(arr == NULL)
+            {
+                return NULL;
+            }
+        arr[i] = temp;
+        temp = strtok(NULL," ");
+        i++;
+    }
+    arr = (char**)realloc(arr,sizeof (char*)*(i+1));
+    arr[i] = 0;
+    return arr;
+
+    free(arr);
+}
+/* 
+ * This function retrieves the first line of array 
+ * and returns it as a pointer
+ */
+
+char * get_first_line(char * string)
+{
+const char ch = '\n';
+    char * temp = strchr(string,ch);
+    int size1 = strlen(string);
+    int size2 = strlen(temp);
+
+    int count = (size1-size2);
+
+    int i = 0;
+
+    char * buffer = NULL;
+
+    while(i <= count-1)
+    {
+        buffer = (char*)realloc(buffer,sizeof (char)*(i+1));
+        buffer[i]=*string;
+        i++;
+        string = string +1;
+    }
+  
+    return (buffer);
+    free(buffer);
+
+}
+
+/*
+ * Will seperate the headers from the inputs and compare them
+ * Will return value of 1 if they match and 0 if
+ * they don't match. Possible error if the vector headers match
+ * but are in different order
+ */
+bool output_vector_headers_comp(char * buffer1, char * buffer2)
+{
+    char *ptr1 = get_first_line(buffer1);
+    char *ptr2 = get_first_line(buffer2);
+
+    char * * ptrptr1 = string_to_token_array(ptr1);
```

Comment:
Use name that have meaning, that gives context to the value it holds, not how it holds it.

Reply:
https://dmitripavlutin.com/coding-like-shakespeare-practical-function-naming-conventions/

These are just little guidelines, they don't exactly reflect how things are done, but the idea is the same.

They use camel case, we use snake case (except in parse_to_ast... The convention should be uniformized at some point )